### DISCUSSION TITLE
results.txt interpretation

### DISCUSSION THREAD

**Discussion by jb-ssncse** [ultralytics/yolov5#2191] [Q&A] [answered]
Hi,
1.Can you please tell, what are the values refer to after image size in results.txt while training
2. I can understand the working of detect.py, but Where to specify path for trained model on custom data in test.py ?

**Comment by glenn-jocher** [comment, marked as answer]
Columns in results.txt are losses (train and val) and metrics.

You pass --weights the same way in all python files, i.e.:
`python test.py --weights /path/to/best.pt`

**Reply by jb-ssncse** [reply]
Thank you for the prompt reply

**Comment by glenn-jocher** [comment]
BTW results.txt is plotted as results.png after training completes, I would recommend you look at results.png rather than trying to understand the txt file.

**Reply by jb-ssncse** [reply]
Thank you, i will look for the png file

**Comment by kashishgoyal31** [comment]
can you please suggest how can I analyse the results.png file for custom dataset. How can I get the explanation of the parameters. How do I know whether the training is file for do I need to retrain the model. Pls reply

**Reply by kashishgoyal31** [reply]
I want explanation of the graphs that are generated while training

**Reply by glenn-jocher** [reply]
@kashishgoyal31 `results.png` displays train and val losses and metrics (P, R, mAP)

![image](https://user-images.githubusercontent.com/26833433/129556555-237f3f7e-79ae-4875-83d7-5201415e71ea.png)

**Reply by kashishgoyal31** [reply]
Could you please discuss in detail what does each of these graphs represent (box_loss, obj_loss, cls_loss)?
Difference between mAP_0.5 and mAP_0.5:0.95??

**Reply by glenn-jocher** [reply]
@kashishgoyal31 YOLO uses 3 loss components, box, objectness and classification. You can read about them in the YOLOv3 paper. https://arxiv.org/abs/1804.02767

For metrics simply do a Google search, i.e. 'object detection metrics'.

**Reply by kashishgoyal31** [reply]
I could not understand why the objectness loss increases in all cases in validation. What does obj_loss signify?

**Comment by kashishgoyal31** [comment]
Sir,

I am facing another issue that when I am training on the custom dataset,
the model is not accepting the images and prompts the error of a corrupted
image, sharing a screenshot for the same. Please provide the solution as
90% of the dataset is not considered.


Regards
Kashish Goyal
